cdap-ui/app/cdap/components/PipelineCanvasActions/PipelineCommentsActionBtn.tsx

```diff
@@ -71,34 +74,44 @@ const useStyle = makeStyles<Theme, Partial<IPipelineCommentsActionBtnProps>>((th
 
 interface IPipelineCommentsActionBtnProps {
   tooltip: string;
-  doesStagesHaveComments: boolean;
-  onCommentsToggle: () => void;
-  toggle: boolean;
+  shouldShowMarker: boolean;
+  onCommentsToggle: (toggle: boolean) => void;
+  onChange: (comments: IPipelineComment[]) => void;
+  comments: IPipelineComment[];
+  disabled?: boolean;
 }
 
 function PipelineCommentsActionBtn({
   tooltip,
-  doesStagesHaveComments,
   onCommentsToggle,
-  toggle,
+  onChange,
+  comments = [],
+  disabled,
 }: IPipelineCommentsActionBtnProps) {
-  const [localToggle, setLocalToggle] = React.useState(toggle);
-  const [showMarker, setShowMarker] = React.useState(doesStagesHaveComments);
+  const [localToggle, setLocalToggle] = React.useState(false);
+  const [showMarker, setShowMarker] = React.useState(comments.length > 0);
+  const [anchorId] = React.useState(`id-${uuidv4()}`);
+  const [anchorEl, setAnchorEl] = React.useState(null);
   const classes = useStyle({ toggle: localToggle });
-  const onClick = () => {
-    setLocalToggle(!toggle);
-    if (typeof onCommentsToggle === 'function') {
-      onCommentsToggle();
-    }
+
+  const onClick = (e) => {
+    setLocalToggle(true);
+    setAnchorEl(e.currentTarget);
+    onCommentsToggle(false);
+  };
+  const onClose = () => {
+    setLocalToggle(false);
+    setAnchorEl(null);
+    onCommentsToggle(true);
   };
+
   React.useEffect(() => {
-    if (typeof doesStagesHaveComments === 'boolean') {
-      setShowMarker(doesStagesHaveComments);
+    if (Array.isArray(comments) && comments.length > 0) {
+      setShowMarker(comments.length > 0);
```

Comment:
Just make this `setShowMarker(true)`? Or even make this whole block `setShowMarker(Array.isArray(comments) && comments.length > 0)`?